app/main/index.ts

```diff
@@ -294,24 +294,39 @@ app.on('ready', () => {
 	ipcMain.on('downloadFile', (_event: Electron.IpcMessageEvent, url: string, downloadPath: string) => {
 		page.downloadURL(url);
 		page.session.once('will-download', (_event: Event, item) => {
-			const filePath = path.join(downloadPath, item.getFilename());
-
-			const getTimeStamp = (): any => {
-				const date = new Date();
-				return date.getTime();
-			};
-
-			const formatFile = (filePath: string): string => {
-				const fileExtension = path.extname(filePath);
-				const baseName = path.basename(filePath, fileExtension);
-				return `${baseName}-${getTimeStamp()}${fileExtension}`;
-			};
-
-			// Update the name and path of the file if it already exists
-
-			const updatedFilePath = path.join(downloadPath, formatFile(filePath));
-
-			const setFilePath = fs.existsSync(filePath) ? updatedFilePath : filePath;
+			let setFilePath: string;
+			let shortFileName: string;
+			if (ConfigUtil.getConfigItem("promptDownload", false)) {
+				const showDialogOptions: object = {
+					defaultPath: path.join(downloadPath, item.getFilename())
+				};
+
+				setFilePath = dialog.showSaveDialog(mainWindow, showDialogOptions);
```

Comment:
I'm not sure why, but this doesn't seem to run immediately on my Windows laptop. I ended up clicking on the same link 5-6 times before I got the save dialog. 
Would it be a good idea to use `showSaveDialogSync` instead?

Reply:
Looks like the version of Electron you are using has saveDialog be the synchronous one. In some later version it seems to have been renamed to saveDialogSync and then a new saveDialog added.